plaid/src/apis/okta/mod.rs

```diff
@@ -1,20 +1,39 @@
 mod groups;
 mod users;
 
+use jwt_simple::{claims::Claims, prelude::{Duration as JwtDuration, RS256KeyPair, RSAKeyPairLike}};
 use reqwest::Client;
-
-use serde::Deserialize;
+use serde::{de, Deserialize, Serialize};
 
 use std::{string::FromUtf8Error, time::Duration};
 
 use super::default_timeout_seconds;
 
+#[derive(Deserialize)]
+#[serde(untagged)]
+enum Authentication {
+    ApiKey { token: String },
+    OktaApp {
+        client_id: String,
+        #[serde(deserialize_with = "private_key_deserializer")]
+        private_key: RS256KeyPair
+    }
+}
+
+fn private_key_deserializer<'de, D>(deserializer: D) -> Result<RS256KeyPair, D::Error>
+where
+    D: de::Deserializer<'de>,
+{
+    let pem_key = String::deserialize(deserializer)?;
+    Ok(RS256KeyPair::from_pem(&pem_key).map_err(|_| de::Error::custom("Could not deserialize app's private key")))?
```

Comment:
@obelisk For what it's worth, if I change my key from
```
private_key = """
-----BEGIN PRIVATE KEY-----
MIIJQgIBADANBgkqhkiG9w0BAQEFAASCCSwwggkoAgEAAoICAQC1c7ImexKmg5Fg
1NbqTVAXDCZ6ytXisAuxBbQDac34PN28JdAN9hNzE2ws/FYQLcBv5zAnq0LNHOJU
...
```

to

```
private_key = """
aaa
-----BEGIN PRIVATE KEY-----
MIIJQgIBADANBgkqhkiG9w0BAQEFAASCCSwwggkoAgEAAoICAQC1c7ImexKmg5Fg
1NbqTVAXDCZ6ytXisAuxBbQDac34PN28JdAN9hNzE2ws/FYQLcBv5zAnq0LNHOJU
...
```
it still works. And I mean, it really works: fetching user profiles and deleting users from Okta groups execute correctly.

Changing it to something like
```
private_key = """
-----BEGIN PRIVATE KEY-----
aaaMIIJQgIBADANBgkqhkiG9w0BAQEFAASCCSwwggkoAgEAAoICAQC1c7ImexKmg5Fg
1NbqTVAXDCZ6ytXisAuxBbQDac34PN28JdAN9hNzE2ws/FYQLcBv5zAnq0LNHOJU
...
```
instead does not work and we obtain

ERROR plaid::config] Encountered parsing error while reading configuration with interpolated secrets!. Error: data did not match any variant of untagged enum Authentication for key `apis.okta.authentication`

So I suppose the parser looks for the header before starting to check(?). Also, minor detail, the custom error above ("Could not deserialize... ") is swallowed by a higher level error.

Reply:
Sounds fine to me, just an intricacy of how the parser for for the key works